src/test/scala/carts/cart/RepositoryTest.scala

```diff
@@ -0,0 +1,177 @@
+import carts.cart.valueobjects.{CartId, Customer, Store}
+import carts.cart.Repository.{CartAlreadyPresent, CartNotFound, OperationFailed}
+
+class RepositoryTest extends AnyFunSpec with TestContainerForAll {
+
+  private val timeout: FiniteDuration = 300.seconds
+
+  override val containerDef: PostgreSQLContainer.Def = PostgreSQLContainer.Def(
+    dockerImageName = DockerImageName.parse("postgres:15.1"),
+    databaseName = "carts",
+    username = "test",
+    password = "test",
+    commonJdbcParams = CommonParams(timeout, timeout, Some("carts.sql"))
+  )
+
+  @SuppressWarnings(Array("org.wartremover.warts.Var", "scalafix:DisableSyntax.var"))
+  private var repository: Option[Repository] = None
+
+  override def afterContainersStart(containers: Containers): Unit =
+    repository = Some(
+      Repository(
+        ConfigFactory
+          .load()
+          .getConfig("ctx")
+          .withValue("dataSource.portNumber", ConfigValueFactory.fromAnyRef(containers.container.getFirstMappedPort.intValue()))
+      )
+    )
+
+  private val cartId: CartId = CartId(15).getOrElse(fail())
+  private val store: Store = Store(200).getOrElse(fail())
+  private val customer: Customer = Customer("addr_3ss!@email.com").getOrElse(fail())
+
+  describe("A locked cart") {
+    describe("after being added to the database") {
+      it("should be present in the database") {
+        val db: Repository = repository.getOrElse(fail())
+        db.add(LockedCart(cartId, store))
+        val cart: Cart = db.findById(cartId, store).getOrElse(fail())
+        cart.cartId shouldBe cartId
+        cart.store shouldBe store
+        cart.isMovable shouldBe false
+        db.remove(cart).getOrElse(fail())
+      }
+    }
+
+    describe("after being added and then deleted from the database") {
+      it("should not be present in the database") {
+        val db: Repository = repository.getOrElse(fail())
+        val lockedCart: Cart = LockedCart(cartId, store)
+        db.add(lockedCart).getOrElse(fail())
+        db.remove(lockedCart).getOrElse(fail())
+        db.findById(cartId, store).left.value shouldBe CartNotFound
+      }
+    }
+
+    describe("if already present in the database") {
+      it("should not allow a new registration") {
+        val db: Repository = repository.getOrElse(fail())
+        val lockedCart: Cart = LockedCart(cartId, store)
+        db.add(lockedCart).getOrElse(fail())
+        db.add(lockedCart).left.value shouldBe CartAlreadyPresent
+        db.remove(lockedCart).getOrElse(fail())
+      }
+    }
+
+    describe("if added with the same cart identifier as another cart, but different store") {
+      it("should be added to the database") {
+        val db: Repository = repository.getOrElse(fail())
+        val c0 = LockedCart(CartId(0).getOrElse(fail()), Store(0).getOrElse(fail()))
+        val c1 = LockedCart(CartId(0).getOrElse(fail()), Store(1).getOrElse(fail()))
```

Comment:
These are not valid scala variable names, prefer something like "firstCart", "secondCart"